### PR TITLE
Refactoring of some `_MAX_PATH` code sections

### PR DESCRIPTION
**Pull request by MarcelHB**
## Description
See #1847.

You still need buffers in some way for Win API but I guess Sonar and Coverty don't panic anymore.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code 
- [x] I have tested the proposed changes
- [x] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

File: platforms/vita/dirent.h [hunk 1]
```diff
@@ -26,50 +26,54 @@
...
-inline dirent* readdir(DIR *dirp)
+inline dirent* readdir(DIR *dir)
 {
 	static dirent de;
 	//vitasdk kind of skips current directory entry..
-	if (dirp->is_first) {
-		dirp->is_first = 0;
+	if (dir->is_first) {
+		dir->is_first = false;
 		strncpy(de.d_name, ".", 1);
```
Comment (bradallred) [inline platforms/vita/dirent.h:62]: shouldn't these `strncpy`s just be an assignment now?

Comment (MarcelHB) [reply platforms/vita/dirent.h:62]: I did not want to touch that a lot (yet I noticed my renaming introduced a mistake ...).

Ok but I've found a ctor of std::string that ought to supersede the 2nd `strncpy`.

Comment (MarcelHB) [reply platforms/vita/dirent.h]: Soft-tested my changes on godbolt

File: gemrb/core/System/VFS.cpp [hunk 2]
```diff
@@ -55,66 +55,68 @@
...
+	// consider $PATH\*.*\0 for _wfindfirst
+	if (strlen(filename) > _MAX_PATH - 5) {
+		return nullptr;
+	}
 
-	mbstowcs(t_filename, filename, _MAX_PATH - 1);
-	StringCbPrintf(dirp->path, _MAX_PATH * sizeof(TCHAR), TEXT("%s%s*.*"), t_filename, TEXT("\\"));
+	std::wstring buffer(_MAX_PATH, L'\0');
+	mbstowcs(const_cast<wchar_t*>(buffer.data()), filename, buffer.length() - 1);
+	dir->path = std::move(fmt::format(L"{}\\*.*", buffer.c_str()));
```
Comment (bradallred) [inline gemrb/core/System/VFS.cpp:92]: `std::move` is redundant here; function returns are already rvalues.

Comment (MarcelHB) [reply gemrb/core/System/VFS.cpp:92]: Right

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**